specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingSubscription.json

```diff
@@ -151,246 +237,509 @@
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/billing/"
+        },
         "operationId": "BillingSubscriptions_Update",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
             "$ref": "./types.json#/parameters/billingAccountName"
           },
           {
-            "$ref": "./types.json#/parameters/subscriptionId"
+            "$ref": "./types.json#/parameters/billingSubscriptionName"
           },
           {
             "$ref": "./types.json#/parameters/apiVersionParameter"
           },
           {
-            "description": "Request parameters that are provided to the update billing subscription operation.",
             "in": "body",
-            "name": "parameters",
-            "required": true,
+            "name": "body",
+            "description": "The billing properties of a subscription.",
             "schema": {
-              "$ref": "#/definitions/BillingSubscription"
+              "$ref": "#/definitions/BillingSubscriptionPatch"
             }
           }
         ],
         "responses": {
           "200": {
-            "description": "OK. The request has succeeded.",
+            "description": "The billing properties of a subscription.",
             "schema": {
               "$ref": "#/definitions/BillingSubscription"
             }
           },
+          "202": {
+            "description": "An operation accepted result that returns a 'Location' header that the client can poll to view the result of the operation.",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
           "default": {
-            "description": "Error response describing why the operation failed.",
+            "description": "Default response",
             "schema": {
-              "$ref": "./types.json#/definitions/ArmError"
+              "$ref": "./types.json#/definitions/ArmErrorResponse"
             }
           }
         },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "BillingSubscriptionsUpdate": {
+            "$ref": "./examples/billingSubscriptionsUpdate.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingSubscriptions/{billingSubscriptionName}/cancel": {
+      "post": {
         "tags": [
-          "BillingSubscriptions"
+          "BillingSubscription"
+        ],
+        "description": "Cancels a usage-based subscription. This operation is supported only for billing accounts of type Microsoft Partner Agreement.",
```

Comment:
I don't think we should have 2 different cancellation patterns... Can we converge on 1?

Reply:
@ramkd12 @amberbhargava to comment here